ibis/common/patterns.py

```diff
@@ -1596,7 +1596,10 @@ def pattern(obj: AnyType) -> Pattern:
     elif isinstance(obj, Mapping):
         return PatternMapping(obj)
     elif isinstance(obj, type):
-        return InstanceOf(obj)
+        if issubclass(obj, Coercible):
+            return CoercedTo(obj)
+        else:
+            return InstanceOf(obj)
```

Comment:
Is this related to testing? If not, can you submit a separate PR after you've moved things around?

Reply:
I'm extracting it to another commit, but the changes in the test files depend on it.